src/components/NewTodoForm/NewTodoForm.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FC, useContext, useState } from 'react';
+import classNames from 'classnames/bind';
+import { getTodos, sendNewTodo, toggleTodoStatus } from '../../api/todos';
+import { AppTodoContext } from '../AppTodoContext/AppTodoContext';
+import { ErrorType } from '../Error/Error.types';
+import { USER_ID } from '../../react-app-env';
+
+export const NewTodoForm: FC = () => {
+  const {
+    todos,
+    setTodos,
+    setTempTodo,
+    setErrorMessage,
+    setVisibleTodos,
+    completedTodos,
+    setProcessingTodoIDs,
+  } = useContext(AppTodoContext);
+
+  const [inputValue, setInputValue] = useState<string>('');
+  const [isInputDisabled, setIsInputDisabled] = useState<boolean>(false);
+
+  const handleInputChange = (value: string) => {
+    setErrorMessage(ErrorType.NoError);
+    setInputValue(value);
+  };
+
+  const handleAddingTodoOnAPI = async (title: string) => {
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    });
+
+    setIsInputDisabled(true);
+    try {
+      const newTodo = await sendNewTodo(inputValue, USER_ID);
+
+      setTodos(await getTodos(USER_ID));
+      setVisibleTodos(
+        prevTodos => [...prevTodos, newTodo],
+      );
+    } catch {
+      setErrorMessage(ErrorType.NewTodoError);
+    } finally {
+      setTempTodo(null);
+      setIsInputDisabled(false);
+    }
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setInputValue('');
+
+    const validatedValue = inputValue.trim();
+
+    if (validatedValue === '') {
+      setErrorMessage(ErrorType.InputError);
+
+      return;
+    }
+
+    setErrorMessage(ErrorType.NoError);
+    handleAddingTodoOnAPI(validatedValue);
+  };
+
+  const handleToggleAll = async () => {
+    setProcessingTodoIDs(completedTodos.map(completedTodo => completedTodo.id));
+
+    completedTodos.forEach(async (completedTodo) => {
+      try {
+        await toggleTodoStatus(completedTodo);
+      } catch {
+        setErrorMessage(ErrorType.UpdateTodoError);
+      } finally {
+        setProcessingTodoIDs([]);
+      }
+
+      const newTodos = await getTodos(USER_ID);
+
+      setTodos(newTodos);
+      setVisibleTodos(newTodos);
+    });
+  };
+
+  return (
+    <>
+      <button
+        aria-label="toggle all active"
+        type="button"
+        className={classNames(
+          'todoapp__toggle-all',
+          { active: todos.length === completedTodos.length },
+        )}
+        onClick={handleToggleAll}
+      />
+
```

Comment:
Instead of using a separate function `handleInputChange`, you can directly call `setInputValue` in the `onChange` event handler. 🚀